### PR TITLE
Update ioredis cluster types

### PR DESCRIPTION
**Pull request by nickroberts**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/luin/ioredis/blob/master/lib/cluster/index.ts#L321>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by nickroberts** [comment]
This fixes #27391

**Review by plantain-00 — CHANGES_REQUESTED** [review]
It is difficult to review because of too much code format change

**Review by nickroberts — COMMENTED** [review]
I removed the extra formatting so it would be easier to review. I use Prettier on pretty much everything, so that was what happened. Thanks!

File: types/ioredis/index.d.ts [hunk 1]
```diff
@@ -873,11 +873,14 @@ declare namespace IORedis {
 
     type ClusterNode = string | number | NodeConfiguration;
 
-    interface Cluster extends NodeJS.EventEmitter, Commander {
-        new(nodes: ClusterNode[], options?: ClusterOptions): Redis;
+    interface Cluster {
```
Comment (plantain-00) [inline types/ioredis/index.d.ts]: It seems `Cluster` still need ` extends NodeJS.EventEmitter, Commander`, or no `cluster.get`: https://github.com/luin/ioredis#cluster

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by nickroberts — COMMENTED** [review]
I went ahead and extended `NodeJS.EventEmitter` and `Commander`. I was going through more of the source code, and it looks like there are more types that need to be updated (like the `sendCommand` function on `Commander`), but I at least wanted to get the `Cluster` type working correctly (for the most part).

**Review by plantain-00 — APPROVED** [review]

File: types/ioredis/index.d.ts [hunk 2]
```diff
@@ -874,10 +874,13 @@ declare namespace IORedis {
     type ClusterNode = string | number | NodeConfiguration;
 
     interface Cluster extends NodeJS.EventEmitter, Commander {
-        new(nodes: ClusterNode[], options?: ClusterOptions): Redis;
         connect(callback: () => void): Promise<any>;
         disconnect(): void;
-        nodes(role: string): Redis[];
+        nodes: Redis[];
```
Comment (SimonSchick) [inline types/ioredis/index.d.ts]: I don't understand this change, in the code you linked this is a function, not a property?

Comment (nickroberts) [reply types/ioredis/index.d.ts]: Good catch! I pushed an update to make it a function that accepts an optional `NodeType`.

**Review by SimonSchick — APPROVED** [review]

**Comment by adamhathcock** [comment]
This PR broke me and others.  Reverting to 4.0.8 fixes it: https://github.com/luin/ioredis/issues/807

**Comment by nickroberts** [comment]
> This PR broke me and others. Reverting to 4.0.8 fixes it: [luin/ioredis#807](https://github.com/luin/ioredis/issues/807)

I'll take a look and try to get another PR in to fix this. There are probably some other functions and properties of Cluster that are available that need to be added.

As a stop gap, you can probably typecast as `any` (I had to do this for 4.0.8).

**Comment by nickroberts** [comment]
I created a new PR to add the get / set methods: #33655

**Pull request merged**